src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php

```diff
@@ -34,6 +34,13 @@ class AuthorizationChecker implements AuthorizationCheckerInterface
 
     public function __construct(TokenStorageInterface $tokenStorage, AuthenticationManagerInterface $authenticationManager, AccessDecisionManagerInterface $accessDecisionManager, bool $alwaysAuthenticate = false, bool $exceptionOnNoToken = true)
     {
+        if (false !== $alwaysAuthenticate) {
+            trigger_deprecation('symfony/security-core', '5.4', 'Not setting the 4th argument of "%s" to "false" is deprecated.', __METHOD__);
+        }
+        if (false !== $exceptionOnNoToken) {
+            trigger_deprecation('symfony/security-core', '5.4', 'Not setting the 5th argument of "%s" to "false" is deprecated.', __METHOD__);
```

Comment:
Why not deprecate setting them completely?

Reply:
Not setting `$exceptionOnNoToken` means you'll fall back to the legacy behavior (it's an opt-in flag). So it must be set to `false`, but we can safely remove the argument in Symfony 6.

Reply:
Thank you.